eslint/eslint-patch/src/_patch-base.ts

```diff
@@ -0,0 +1,202 @@
+        const eslintrcFolder = path.dirname(
+          require.resolve('@eslint/eslintrc/package.json', {
+            paths: [currentModule.path]
+          })
+        );
+
+        if (path.join(eslintrcFolder, '/lib/config-array-factory.js') == currentModule.filename) {
+          configArrayFactoryPath = path.join(eslintrcFolder, 'lib/config-array-factory.js');
+          moduleResolverPath = path.join(eslintrcFolder, 'lib/shared/relative-module-resolver');
+          namingPath = path.join(eslintrcFolder, 'lib/shared/naming');
+        }
+      } catch (ex: unknown) {
+        // Module resolution failures are expected, as we're walking
+        // up our require stack to look for eslint. All other errors
+        // are rethrown.
+        if (!isModuleResolutionError(ex)) {
+          throw ex;
+        }
+      }
+    } else {
+      // Next look for a file in ESLint's folder
+      //   .../eslint/lib/cli-engine/cli-engine.js
+      try {
+        const eslintCandidateFolder = path.dirname(
+          require.resolve('eslint/package.json', {
+            paths: [currentModule.path]
+          })
+        );
+
+        if (path.join(eslintCandidateFolder, 'lib/cli-engine/cli-engine.js') == currentModule.filename) {
+          eslintFolder = eslintCandidateFolder;
+          break;
+        }
+      } catch (ex: unknown) {
+        // Module resolution failures are expected, as we're walking
+        // up our require stack to look for eslint. All other errors
+        // are rethrown.
+        if (!isModuleResolutionError(ex)) {
+          throw ex;
+        }
+      }
+    }
+
+    if (!currentModule.parent) {
+      break;
+    }
+    currentModule = currentModule.parent;
+  }
+}
+
+if (!eslintFolder) {
+  // Probe for the <7.8.0 layout:
+  for (let currentModule = module; ; ) {
+    // For ESLint <7.8.0, config-array-factory.js was at this path:
+    //   .../eslint/lib/cli-engine/config-array-factory.js
+    if (/[\\/]eslint[\\/]lib[\\/]cli-engine[\\/]config-array-factory\.js$/i.test(currentModule.filename)) {
+      eslintFolder = path.join(path.dirname(currentModule.filename), '../..');
+      configArrayFactoryPath = path.join(eslintFolder, 'lib/cli-engine/config-array-factory');
+      moduleResolverPath = path.join(eslintFolder, 'lib/shared/relative-module-resolver');
+      namingPath = path.join(eslintFolder, 'lib/shared/naming');
+      break;
+    }
+
+    if (!currentModule.parent) {
+      // This was tested with ESLint 6.1.0 .. 7.12.1.
+      throw new Error(
+        'Failed to patch ESLint because the calling module was not recognized.\n' +
+          'If you are using a newer ESLint version that may be unsupported, please create a GitHub issue:\n' +
+          'https://github.com/microsoft/rushstack/issues'
+      );
+    }
+    currentModule = currentModule.parent;
+  }
+}
+
+// Detect the ESLint package version
+const eslintPackageJson = fs.readFileSync(path.join(eslintFolder, 'package.json')).toString();
```

Comment:
Does calling `toString()` provide some advantage over specifying the encoding in the `readFileSync` call? I'd expect the latter to potentially optimize internally for different buffering behavior.